src/test/java/uk/gov/ons/census/action/schedule/ActionRuleProcessorIT.java

```diff
@@ -0,0 +1,156 @@
+import static com.github.tomakehurst.wiremock.client.WireMock.verify;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import java.io.IOException;
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import org.jeasy.random.EasyRandom;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import uk.gov.ons.census.action.messaging.RabbitQueueHelper;
+import uk.gov.ons.census.action.model.dto.PrintFileDto;
+import uk.gov.ons.census.action.model.dto.UacQidDTO;
+import uk.gov.ons.census.action.model.entity.ActionPlan;
+import uk.gov.ons.census.action.model.entity.ActionRule;
+import uk.gov.ons.census.action.model.entity.ActionType;
+import uk.gov.ons.census.action.model.entity.Case;
+import uk.gov.ons.census.action.model.repository.ActionPlanRepository;
+import uk.gov.ons.census.action.model.repository.ActionRuleRepository;
+import uk.gov.ons.census.action.model.repository.CaseRepository;
+import uk.gov.ons.census.action.model.repository.UacQidLinkRepository;
+
+@ContextConfiguration
+@SpringBootTest
+@ActiveProfiles("test")
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+@RunWith(SpringJUnit4ClassRunner.class)
+public class ActionRuleProcessorIT {
+
+  private static final int DELAY_ACTION_BY_SECONDS = 5;
+
+  @Value("${queueconfig.outbound-printer-queue}")
+  private String outboundPrinterQueue;
+
+  @Autowired private RabbitQueueHelper rabbitQueueHelper;
+  @Autowired private CaseRepository caseRepository;
+  @Autowired private UacQidLinkRepository uacQidLinkRepository;
+  @Autowired private ActionRuleRepository actionRuleRepository;
+  @Autowired private ActionPlanRepository actionPlanRepository;
+  @Autowired private ActionRuleProcessor actionRuleProcessor;
+  private static final EasyRandom easyRandom = new EasyRandom();
+
+  @Value("${caseapi.port}")
+  private int caseApiPort;
+
+  @Value("${caseapi.host}")
+  private String caseApiHost;
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  @Rule public WireMockRule mockCaseApi = new WireMockRule(wireMockConfig().port(8089));
```

Comment:
Minor comment but is it worth adding a setUp method to purge the queues etc here? [Similar to this](https://github.com/ONSdigital/census-rm-case-processor/blob/8959ea6e147d7235b0bed94538d3a6a076248439/src/test/java/uk/gov/ons/census/casesvc/messaging/InvalidAddressReceiverIT.java#L57-L64)